neural_modelling/src/neuron/synapses.c

```diff
@@ -382,3 +382,28 @@ uint32_t synapses_get_pre_synaptic_events(void) {
     return (num_fixed_pre_synaptic_events +
             synapse_dynamics_get_plastic_pre_synaptic_events());
 }
+
+void synapses_flush_ring_buffers(void) {
+    uint32_t n_neurons_power_2 = n_neurons;
+    uint32_t log_n_neurons = 1;
+    if (n_neurons != 1) {
+        if (!is_power_of_2(n_neurons)) {
+            n_neurons_power_2 = next_power_of_2(n_neurons);
+        }
+        log_n_neurons = ilog_2(n_neurons_power_2);
+    }
+
+    uint32_t n_synapse_types_power_2 = n_synapse_types;
+    if (!is_power_of_2(n_synapse_types)) {
+        n_synapse_types_power_2 = next_power_of_2(n_synapse_types);
+    }
+    uint32_t log_n_synapse_types = ilog_2(n_synapse_types_power_2);
+
+    uint32_t n_ring_buffer_bits =
+            log_n_neurons + log_n_synapse_types + SYNAPSE_DELAY_BITS;
+    uint32_t ring_buffer_size = 1 << (n_ring_buffer_bits);
```

Comment:
This does a lot of work to get the ring buffer size!  There are two easy options to avoid this:
 1. Store ```ring_buffer_size``` where it is calculated previously in a static global variable.
 1. See lines 304-326 above for how to loop over the neuron and synapse indices and use that to get the ring buffer index.

I suspect the first is the easier...

Reply:
Done - stored as static global.